src/basic-languages/typescript/typescript.ts

```diff
@@ -222,143 +131,249 @@ export const language = {
+
+		{ include: 'common' },
+	  ],
+
+	  common: [
+
+		// identifiers and keywords
+		[/(#?[a-zA-Z_$][\w$]*)([<(]?)/, [
+		  {
+			cases: {
+			  '@typeKeywords': 'type.identifier',
+			  '@alwaysKeyword': 'keyword',
+			  '$1~#?[A-Z].*': 'type.identifier',
+			  '$2': 'method',
+			  '@ctrlKeywords': 'keyword.flow',
+			  '@keywords': 'keyword',
+			  '@default': 'identifier',
+			}
+		  },
+		  {
+			cases: {
+			  '$2==<': { token: '@brackets', next: '@typeparams' },
+			  '@default': '@rematch',
+			}
+		  },
+		]],
+
+		// whitespace
+		{ include: '@whitespace' },
 
-		string_double: [
-			[/[^\\"]+/, 'string'],
-			[/@escapes/, 'string.escape'],
-			[/\\./, 'string.escape.invalid'],
-			[/"/, 'string', '@pop']
+		// regular expression: ensure it is terminated before beginning (otherwise it is an opeator)
+		[
+		  /\/(?=([^\\\/]|\\.)+\/([dgimsuy]*)(\s*)(\.|;|,|\)|\]|\}|$))/,
+		  { token: 'regexp', bracket: '@open', next: '@regexp' }
 		],
 
-		string_single: [
-			[/[^\\']+/, 'string'],
-			[/@escapes/, 'string.escape'],
-			[/\\./, 'string.escape.invalid'],
-			[/'/, 'string', '@pop']
+		// delimiters and operators
+		[/[()\[\]]/, '@brackets'],
+		[/[<>](?!@symbols)/, '@brackets'],
+		[/!(?=([^=]|$))/, 'delimiter'],
+		[
+		  /@symbols/,
+		  {
+			cases: {
+			  '@operators': 'delimiter',
+			  '@default': ''
+			}
+		  }
 		],
 
-		string_backtick: [
-			[/\$\{/, { token: 'delimiter.bracket', next: '@bracketCounting' }],
-			[/[^\\`$]+/, 'string'],
-			[/@escapes/, 'string.escape'],
-			[/\\./, 'string.escape.invalid'],
-			[/`/, 'string', '@pop']
+		[/\.\.\./, 'keyword'],
+
+		// numbers
+		[/(@digits)[eE]([\-+]?(@digits))?/, 'number.float'],
+		[/(@digits)\.(@digits)([eE][\-+]?(@digits))?/, 'number.float'],
+		[/0[xX](@hexdigits)n?/, 'number.hex'],
+		[/0[oO]?(@octaldigits)n?/, 'number.octal'],
+		[/0[bB](@binarydigits)n?/, 'number.binary'],
+		[/(@digits)n?/, 'number'],
+
+		// delimiter: after number because of .\d floats
+		[/[;,.]/, 'delimiter'],
+
+		// strings
+		[/"([^"\\]|\\.)*$/, 'string.invalid'], // non-teminated string
+		[/'([^'\\]|\\.)*$/, 'string.invalid'], // non-teminated string
+		[/"/, 'string', '@string_double'],
+		[/'/, 'string', '@string_single'],
+		[/`/, 'string', '@string_backtick']
+	  ],
+
+	  typeparams: [
+		[/>/, '@brackets', '@pop'],
+		{ include: 'common' }
+	  ],
+
+	  jsxReady: [
+		[/<>/, 'delimiter.html', '@jsxText.FRAGMENT'],
+		[/(<)([A-Z][\w$]*\s*(?:,|extends|implements))/, ['@brackets', { token: '@rematch', next: '@typeparams' }]],
+		[/(<)(\s*)([\w$])/, ['delimiter.html', '',
+		  { token: '@rematch', next: '@jsxIdent.jsxOpen.' },
+		]],
+	  ],
+
+	  jsxIdent: [
+		[/\./, { token: 'delimiter', switchTo: '$S0^' }],
+		[/[A-Z][\w$]*/, { token: 'type.identifier', switchTo: '$S0$0' }],
+		[/[\w$-]+/, { token: 'tag', switchTo: '$S0$0' }],
+		[/.+/, { token: '@rematch', switchTo: '@$S2.$S3.$S4' }],
+	  ],
+
+	  jsxOpen: [
+		[/{/, { token: 'keyword', next: '@root.INJSX', bracket: '@open' }],
```

Comment:
Hi @thesoftwarephilosopher, Thank you so much for sharing this token provider file. It's very helpful and straightforward, and it solved my problem very easily!

I noticed a small inconsistency where the first and second brackets had different tokens in some cases. I made a slight modification to make them consistent.
**Before:**
<img width="581" alt="image" src="https://github.com/user-attachments/assets/fa584b3d-a157-43d8-b024-a2b174c931a4">
<img width="543" alt="image" src="https://github.com/user-attachments/assets/086c4dce-0007-45b5-b411-f7865e73e5c7">
**After:**
<img width="550" alt="image" src="https://github.com/user-attachments/assets/beb39425-a702-44f3-a918-86c1f70d503a">
<img width="359" alt="image" src="https://github.com/user-attachments/assets/1a35c4e2-8634-40f3-bcf4-b63f1fbda17a">


```suggestion
		[/{/, { token: 'delimiter.curly', next: '@root.INJSX', bracket: '@open' }],
```